File: src/opentimelineio/indexStreamAddress.cpp
// SPDX-License-Identifier: Apache-2.0
// Copyright Contributors to the OpenTimelineIO project

#include "opentimelineio/indexStreamAddress.h"

namespace opentimelineio { namespace OPENTIMELINEIO_VERSION_NS {

IndexStreamAddress::IndexStreamAddress(int64_t index)
    : Parent()
    , _index(index)
{}

IndexStreamAddress::~IndexStreamAddress()
{}

bool
IndexStreamAddress::read_from(Reader& reader)
{
    return reader.read("index", &_index) && Parent::read_from(reader);
}

void
IndexStreamAddress::write_to(Writer& writer) const
{
    Parent::write_to(writer);
    writer.write("index", _index);
}

}} // namespace opentimelineio::OPENTIMELINEIO_VERSION_NS

File: src/py-opentimelineio/opentimelineio/schema/index_stream_address.py
# SPDX-License-Identifier: Apache-2.0
# Copyright Contributors to the OpenTimelineIO project

from .. core._core_utils import add_method
from .. import _otio


@add_method(_otio.IndexStreamAddress)
def __str__(self):
    return "IndexStreamAddress({})".format(self.index)


@add_method(_otio.IndexStreamAddress)
def __repr__(self):
    return "otio.schema.IndexStreamAddress(index={})".format(repr(self.index))
